content/docs/command-reference/exp/branch.md

```diff
@@ -0,0 +1,43 @@
+# exp branch
+
+Commit the results from an [experiment](/doc/command-reference/exp) in a new Git
+branch, which will become the <abbr>workspace</abbr>.
```

Comment:
The more we document experiments, the more I think it's better to be precise, even if it exposes the complexity of the implementation, rather than invent terminology to avoid those details (for example, results from an experiment are already committed, which makes this description seem misleading).

Would it be more accurate to say something like, "Create a new branch with the commits for an experiment" or "Persist commits for an experiment in a new branch"?

Reply:
I thought I saw a reference to experiments as "hidden branches" somewhere else. If we use that terminology, can we just describe `exp branch` as making the hidden branch visible/persistent?

Reply:
> results from an experiment are already committed

True. But not in a regular Git branch (branches technically are Git refs in `refs/heads`). And we don't really say experiments are commits except in a collapsed details section of `exp run`. We actually say they help you avoid polluting the repo with too many commits/branches.

> saw a reference to experiments as "hidden branches" somewhere

Nope (maybe in a previous iteration). We have "custom Git branch" in the Checkpoints collapsed details section somewhere.

Reply:
> "Create a new branch with the commits for an experiment
> Persist commits for an experiment in a new branch

"commits for an experiment" lacks context (we don't expect most users to know how exps are implemented internally). And also without enough context, "to persist" means something else (being persistent) but I did link use that term in the description now, linked to the corresponding section of the Exps guide.

p.s. BTW "which will become the workspace" and other phrases were wrong. That has been removed now 👍

Reply:
Thoughts on replacing "Commit" with anything else: "Add," "Keep," etc.? I don't want to hold this up longer, so go with whatever you prefer, but I don't like saying that `exp branch` commits something when that's not true. I worry it's going to cause additional confusion for anyone who starts to look any deeper.

My larger suggestion of being more direct and precise about what experiments are doing would likely require changes across lots of docs at this point, so I'll keep it as a note for now and maybe open an issue. I wish we could do A/B testing on our docs 😄 .

Reply:
Last suggestion (I hope): "Creates a new branch with the results from an experiment."

Reply:
Don't worry we can discuss as much as needed (even after merged).

> replacing "Commit" with anything else: "Add," "Keep," etc... I don't like saying that exp branch commits something when that's not true.

OK I see your concern is about correctness. But I think it is correct when the experiment in question is based on some other branch (not `HEAD`). AFAU the resulting branch here is always based off `HEAD` which implies that its commits can be new (like git-cherry-pick`).

Will need to try this to figure out the best explanation, and circle back (probably in another iteration) ⌛